Tutorial : Downsampling and Unlimited Metric Retention for Prometheus. (#2801)
thanos-io/thanos: tutorials/katacoda/thanos/2-lts/step1.md

# Step 1 - Start initial Prometheus servers

In this tutorial, we will be using three Prometheus Servers.

> 1. We have one Prometheus server in some eu1 cluster.

Reviewer: ```suggestion
1. We have one Prometheus server in eu1 region.
```

> 2. We have 2 replica Prometheus servers in some us1 cluster that scrapes the same targets.

Reviewer: ```suggestion
2. We have 2 Prometheus servers scraping the same target in us1 region,
```

</pre>

<pre class="file" data-filename="prometheus1_us1.yml" data-target="replace">
global:
  scrape_interval: 15s
  evaluation_interval: 15s
  external_labels:
    cluster: us1
    replica: 1

scrape_configs:
  - job_name: 'prometheus'
    static_configs:
      - targets: ['127.0.0.1:9091','127.0.0.1:9092']
</pre>

## Starting Prometheus Instances

Let's now start three containers representing our three different Prometheus instances.

Execute following commands:

### Prepare "persistent volumes"

```
mkdir -p prometheus0_eu1_data prometheus0_us1_data prometheus1_us1_data
```{{execute}}

### Deploying "EU1"

```
docker run -d --net=host --rm \
    -v $(pwd)/prometheus0_eu1.yml:/etc/prometheus/prometheus.yml \
    -v $(pwd)/prometheus0_eu1_data:/prometheus \
    -u root \
    --name prometheus-0-eu1 \
    quay.io/prometheus/prometheus:v2.14.0 \
    --config.file=/etc/prometheus/prometheus.yml \
    --storage.tsdb.path=/prometheus \
    --web.listen-address=:9090 \
    --web.external-url=https://[[HOST_SUBDOMAIN]]-9090-[[KATACODA_HOST]].environments.katacoda.com \
    --web.enable-lifecycle \
    --web.enable-admin-api && echo "Prometheus EU1 started!"
```{{execute}}

> NOTE: We are using the latest Prometheus image so we can take profit from the latest remote read protocol.

Reviewer: This note will go out of date soon, so lets remove it
